pkg/verifier/aws/aws_verifier.go

```diff
@@ -340,10 +340,33 @@ func (a *AwsVerifier) createEC2Instance(input createEC2InstanceInput) (string, e
 	// Wait up to 5 minutes for the instance to be running
 	waiter := ec2.NewInstanceRunningWaiter(a.AwsClient)
 	if err := waiter.Wait(input.ctx, &ec2.DescribeInstancesInput{InstanceIds: []string{instanceID}}, 2*time.Minute); err != nil {
+
+		resp, err := a.AwsClient.DescribeInstances(context.TODO(), &ec2.DescribeInstancesInput{
+			InstanceIds: []string{instanceID},
+		})
+		if err != nil {
+			fmt.Println("Warning: Waiter Describe instances failure.")
+		}
+
+		var stateCode, stateMessage string
+		if resp.Reservations[0].Instances[0].StateReason.Code != nil {
+			stateCode = *resp.Reservations[0].Instances[0].StateReason.Code
+		}
+		if resp.Reservations[0].Instances[0].StateReason.Message != nil {
+			stateMessage = *resp.Reservations[0].Instances[0].StateReason.Message
+		}
+
+		waiterErr := fmt.Errorf("%s: terminated %s after timing out waiting for instance to be running", err, instanceID)
+
+		if stateCode == invalildKMSCode {
+			waiterErr = fmt.Errorf("Instance %s terminated because of: %s. ", instanceID, stateMessage)
```

Comment:
After reviewing the OSD ticket, it calls out we need to reach out to OCM and others to inform them of this new error message. So we should be certain of the wording here, and check with how they handle current errors (do they string match? do they wanted a typed error?) cc @abyrne55

Reply:
we returning exactly the same as before. I don't think this changes how OCM handles the error. I believe the communication is after the merge is done to communicate with the OCM team that there is this new feature. @abyrne55  Input?

Reply:
It's more that under this specific failure scenario (KMS key access issues) we return a more descriptive error message now than the generic failure message before. If they want to bubble that up nicely to the customer in the API and UI, then how we implement it matters.

Reply:
As discussed in Slack, here is a sample of how we could handle this to make it easier for OCM to know it was a KMS failure and provide a more helpful error message to customers using the API or Console. https://github.com/joshbranham/osd-network-verifier/pull/1